src/query/functions-v2/src/scalars/array.rs

```diff
@@ -124,4 +127,96 @@ pub fn register(registry: &mut FunctionRegistry) {
             }
         ),
     );
+
+    registry.register_passthrough_nullable_3_arg::<EmptyArrayType, UInt64Type, UInt64Type, EmptyArrayType, _, _>(
+        "array_slice",
+        FunctionProperty::default(),
+        |_, _, _| None,
+        vectorize_with_builder_3_arg::<EmptyArrayType, UInt64Type, UInt64Type, EmptyArrayType>(
+            |_, _, _, output, _| {
+                *output += 1;
+                Ok(())
+            }
+        ),
+    );
+
+    registry.register_passthrough_nullable_3_arg::<ArrayType<GenericType<0>>, UInt64Type, UInt64Type, ArrayType<GenericType<0>>, _, _>(
+        "array_slice",
+        FunctionProperty::default(),
+        |_, _, _| None,
+        vectorize_with_builder_3_arg::<ArrayType<GenericType<0>>, UInt64Type, UInt64Type, ArrayType<GenericType<0>>>(
+            |arr, start, end, output, _| {
+                let start = start as usize;
+                let end = end as usize;
+                if arr.len() == 0 || arr.len() <= start || start >= end {
+                    output.push_default();
+                } else {
+                    let range = if end < arr.len() {
+                        Range { start, end }
+                    } else {
+                        Range { start, end: arr.len() }
+                    };
+                    let arr_slice = arr.slice(range);
+                    output.push(arr_slice);
+                }
+                Ok(())
+            }
+        ),
+    );
+
+    registry.register_passthrough_nullable_1_arg::<EmptyArrayType, EmptyArrayType, _, _>(
+        "array_pop_front",
```

Comment:
the new expression framework allows it be named as pop_front (without the array prefix). This could make it more concise but also different from snowflake.

Reply:
> the new expression framework allows it be named as pop_front (without the array prefix).

Does this means we can apply `pop_front`, `pop_back` operations on different data types?

Reply:
besides, its behavior seems more properly be remove_first.

Reply:
I think we are the same as duckdb.

Reply:
the 'pop' brings the confusion of whether the item or the array is returned.

Reply:
rename as `remove_first` and `remove_last`